src/renderer/components/execute/conditions/ConditionForm.tsx

```diff
@@ -0,0 +1,280 @@
+  RowStringOperations,
+  TableGenericOperations,
+  UnitTestOperations,
+} from 'db/entity/enum';
+import { useEffect, useState } from 'react';
+import { Button, Dropdown, Form, InputGroup } from 'react-bootstrap';
+import { Condition } from 'renderer/types';
+
+import '../../../scss/Condition.scss';
+
+interface IConditionErrors {
+  testItem?: boolean;
+  condition?: boolean;
+  compare?: boolean;
+  valueToCheck?: boolean;
+}
+
+interface IConditionProps {
+  addCondition: (c: Condition) => void;
+  table: string;
+}
+
+const ConditionForm = ({ addCondition, table }: IConditionProps) => {
+  const [form, setForm] = useState<Condition>({ table });
+  const [errors, setErrors] = useState<IConditionErrors>({});
+  const [opValues, setOpValues] = useState<string[]>([]);
+  const [columns, setColumns] = useState<DBColumn[]>([]);
+
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const setField = (field: string, value: any) => {
+    setForm({
+      ...form,
+      [field]: value,
+    });
+  };
+
+  // fix error checking for form creation
+  // const validateForm = () => {
+  //   const newErrors: IConditionErrors = {};
+  //   if (!testItem) {
+  //     newErrors.testItem = true;
+  //   }
+
+  //   if (!condition) {
+  //     newErrors.condition = true;
+  //   }
+
+  //   // if (
+  //   //   (operationType === UnitTestOperations.COL || condition === 'Count') && !valueToCheck
+  //   // ) {
+  //   //   newErrors.valueToCheck = true;
+  //   // }
+
+  //   if (condition === 'Count' && !compare) {
+  //     newErrors.compare = true;
+  //   }
+
+  //   return newErrors;
+  // };
+
+  const clearErrors = () => {
+    setErrors({});
+  };
+
+  const clearForm = () => {
+    setForm({});
+  };
+
+  const handleAdd = () => {
+    // const formErrors = validateForm();
+    // if (Object.keys(formErrors).length > 0) {
+    //   setErrors(formErrors);
+    //   return;
+    // }
+
+    addCondition(form);
+
+    clearForm();
+    clearErrors();
+  };
+
+  const updateLevel = (level: UnitTestOperations) => {
+    setForm({ table, level });
+
+    switch (level) {
+      case UnitTestOperations.TableGenericOperations:
+        setOpValues(Object.values(TableGenericOperations));
+        break;
+      case UnitTestOperations.RowBooleanOperations:
+        setOpValues(Object.values(RowBooleanOperations));
+        break;
+      case UnitTestOperations.RowIDOperations:
+        setOpValues(Object.values(RowIDOperations));
+        break;
+      case UnitTestOperations.RowNumberOperations:
+        setOpValues(Object.values(RowNumberOperations));
+        break;
+      case UnitTestOperations.RowStringOperations:
+        setOpValues(Object.values(RowStringOperations));
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    const getColumns = async () => {
+      const cols = await window.procedures.ipcRenderer.fetchColumns(table);
+      setColumns(cols);
+    };
+    getColumns();
```

Comment:
```suggestion
    clearForm();
    clearErrors();
    getColumns();
```
So operation isn't maintained when switching table if options are already selected